test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +94,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
```

Comment:
I think there should be a specific check to see if the command is a mutating command first. Then this check makes sense from a test perspective.